paasta_tools/cleanup_tron_namespaces.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+# Copyright 2015-2018 Yelp Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Usage: ./cleanup_tron_namespaces.py [options]
+
+Delete namespaces that aren't configured in SOA configs for a Tron cluster.
+
+Gets the list of namespaces from Tron, then compares to the namespaces
+defined in SOA configs.
+
+- -d <SOA_DIR>, --soa-dir <SOA_DIR>: Specify a SOA config dir to read from
+"""
+import argparse
+import sys
+
+from paasta_tools import tron_tools
+from paasta_tools.utils import paasta_print
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Cleans up stale Tron namespaces.')
+    parser.add_argument(
+        '-d', '--soa-dir', dest='soa_dir', metavar='SOA_DIR',
+        default=tron_tools.DEFAULT_SOA_DIR,
+        help='Use a different soa config directory',
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = parse_args()
+
+    client = tron_tools.get_tron_client()
+    namespaces = client.list_namespaces()
+    expected_namespaces = tron_tools.get_tron_namespaces_for_cluster(soa_dir=args.soa_dir)
```

Comment:
This script looks risky to me.
Can you make a note to add an option to dry run so we can learn to trust it? Otherwise the first time we run this script it will be very scary.